vxquery-core/src/main/java/org/apache/vxquery/hdfs2/HDFSFunctions.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.vxquery.hdfs2;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Properties;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.LocatedFileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.RemoteIterator;
+
+public class HDFSFunctions {
+
+    private Configuration conf;
+    private FileSystem fs;
+    private String conf_path;
+
+    /**
+     * Create the configuration and add the paths for core-site and hdfs-site as resources.
+     * Initialize an instance of HDFS FileSystem for this configuration.
+     * 
+     * @param hadoop_conf_filepath
+     */
+    public HDFSFunctions() {
+        if (locateConf()) {
+            this.conf = new Configuration();
+
+            conf.addResource(new Path(this.conf_path + "/core-site.xml"));
+            conf.addResource(new Path(this.conf_path + "/hdfs-site.xml"));
+            try {
+                fs = FileSystem.get(conf);
+            } catch (IOException ex) {
+                System.err.println(ex);
+            }
+        } else {
+            System.err.println("Could not locate hdfs configuarion folder.");
+        }
+    }
+
+    /**
+     * Returns true if the file path exists or it is located somewhere in the home directory of the user that called the function.
+     * Searches in subdirectories of the home directory too.
+     * 
+     * @param filename
+     * @return
+     */
+    public boolean isLocatedInHDFS(String filename) {
+        try {
+            //search file path
+            if (fs.exists(new Path(filename))) {
+                return true;
+            }
+        } catch (IOException ex) {
+            System.err.println(ex);
+        }
+        //Search every file and folder in the home directory
+        if (searchInDirectory(fs.getHomeDirectory(), filename) != null) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Searches the given directory and subdirectories for the file.
+     * 
```

Comment:
Yes it does the same, I tested it with the ghcnd folder from the unit tests and works correctly.